db/rdbms/src/main/java/io/camunda/db/rdbms/sql/typehandler/WildcardTransformingStringTypeHandler.java

```diff
@@ -15,36 +15,44 @@
 
 public class WildcardTransformingStringTypeHandler extends StringTypeHandler {
 
+  // Regular expression to match unescaped wildcards
+  public static final String REGEX_ASTERISK = "(?<!\\\\)\\*";
+  public static final String REGEX_QUESTION_MARK = "(?<!\\\\)\\?";
+  public static final String REGEX_PERCENT = "(?<!\\\\)%";
+  public static final String REGEX_UNDERSCORE = "(?<!\\\\)_";
+  public static final Pattern PATTERN_ASTERISK = Pattern.compile(REGEX_ASTERISK);
+  public static final Pattern PATTERN_QUESTION_MARK = Pattern.compile(REGEX_QUESTION_MARK);
+  public static final Pattern PATTERN_PERCENT = Pattern.compile(REGEX_PERCENT);
+  public static final Pattern PATTERN_UNDERSCORE = Pattern.compile(REGEX_UNDERSCORE);
+
   @Override
   public void setNonNullParameter(
       final PreparedStatement ps, final int i, final String parameter, final JdbcType jdbcType)
       throws SQLException {
     ps.setString(i, transformElasticsearchToSql(parameter));
   }
 
-  public static String transformElasticsearchToSql(final String elasticsearchQuery) {
-    // Regular expression to match unescaped wildcards
-    final var regexAsterisk = "(?<!\\\\)\\*";
-    final var regexQuestionMark = "(?<!\\\\)\\?";
-
-    // Replace unescaped * with %
-    final var patternAsterisk = Pattern.compile(regexAsterisk);
-    final var matcherAsterisk = patternAsterisk.matcher(elasticsearchQuery);
-    final var sqlQuery = new StringBuilder();
-    while (matcherAsterisk.find()) {
-      matcherAsterisk.appendReplacement(sqlQuery, "%");
+  private static String replace(
+      final Pattern pattern, final String input, final String replacement) {
+    final var matcher = pattern.matcher(input);
+    final var builder = new StringBuilder();
+    while (matcher.find()) {
+      matcher.appendReplacement(builder, replacement);
     }
-    matcherAsterisk.appendTail(sqlQuery);
+    matcher.appendTail(builder);
+    return builder.toString();
+  }
 
+  public static String transformElasticsearchToSql(String exp) {
```

Comment:
🔧 The name of this method is a bit misleading, I believe. Are we replacing API wildcards with SQL wildcards? If so, let's name it according to that. The fact that ES is using the same wildcard notion as our API is simply a "coincidence".

Reply:
Renamed to `transformParameter`.